src/compute/src/sink/copy_to_s3_oneshot.rs

```
@@ -66,73 +97,24 @@ where
-            for input_frontier in frontiers {
-                frontier.extend(input_frontier.frontier().iter().copied());
-            }
-
-            ok_input.for_each(|_, data| {
-                data.swap(&mut ok_buf);
-                let count: u64 = ok_buf.drain(..).len().try_into().expect("usize into u64");
-                // TODO (mouli): write these to s3 in subsequent PRs
-                row_count = row_count + count;
-            });
-            // TODO (mouli): handle errors
-            if PartialOrder::less_equal(&up_to, &frontier) {
-                finished = true;
-                // We are done, send the final count.
-                if let Some(response_protocol) = response_protocol_handle.borrow_mut().deref_mut() {
-                    response_protocol.send_final_count(row_count);
-                }
-            }
-        }
-    });
-}
-
 /// A type that guides the transmission of number of rows back to the coordinator.
 struct ResponseProtocol {
     pub sink_id: GlobalId,
     pub response_buffer: Option<Rc<RefCell<Vec<(GlobalId, CopyToResponse)>>>>,
 }
 
 impl ResponseProtocol {
-    fn send_final_count(&mut self, rows: u64) {
-        let buffer = self.response_buffer.as_mut().expect("Copy response buffer");
-
-        buffer
-            .borrow_mut()
-            .push((self.sink_id, CopyToResponse::RowCount(rows)));
-
+    // This method should only be called once.
+    fn send(&mut self, count: Result<u64, String>) {
         // The dataflow's input has been exhausted, clear the channel,
         // to avoid sending `CopyToResponse::Dropped`.
-        self.response_buffer = None;
+        if let Some(buffer) = self.response_buffer.take() {
+            let response = match count {
+                Ok(count) => CopyToResponse::RowCount(count),
+                Err(error) => CopyToResponse::Error(error),
+            };
+            buffer.borrow_mut().push((self.sink_id, response));
+        }
```

Before we panicked when the `response_buffer` was already taken, now we silently ignore it. I think we still want to panic, or at least soft-panic, since this method being called twice is a bug.